src/com/hms/flexyosisoftconnector/PayloadManager.java

```diff
@@ -0,0 +1,157 @@
+   */
+  OsisoftJsonPayload osisoftPayload;
+
+  /**
+   * Constant for status okay
+   */
+  static final int STATUS_OKAY = 4;
+
+  /**
+   * Constant for status stop
+   */
+  static final int STATUS_STOP = 5;
+
+  /**
+   * Current payload manager status
+   */
+  private int status = STATUS_OKAY;
+
+  /**
+   * Constructor for the Payload Manager.
+   *
+   * @param communicationType The communication type can be OMF or the old PIWebApi format.
+   **/
+  PayloadManager(int communicationType) {
+
+    // The unfinished payload information is stored in this.
+    osisoftPayload = new OsisoftJsonPayload(communicationType);
+
+    // The complete payload strings are stored in this.
+    payloads = Collections.synchronizedList(new ArrayList());
+  }
+
+  /**
+   * Append a new datapoint to a payload with the specified information.
+   *
+   * @param dataPoint holds all the information for a new datapoint.
+   */
+  public void appendDataPointToPayLoad(DataPoint dataPoint) {
+
+    osisoftPayload.appendDataPointToPayLoad(dataPoint);
+    checkPayloadStatus();
+  }
+
+  /**
+   * Check the status of the payload at the specified index. If it is a complete payload, add it to
+   * the list of start new payload.
+   */
+  private void checkPayloadStatus() {
+    if (osisoftPayload.statusIsComplete()) {
+      addToQueue(osisoftPayload.getPayload());
+      osisoftPayload.startPayload();
+    }
+  }
+
+  /**
+   * Add the specified payload string to the payload queue list.
+   *
+   * @param payload payload to add to queue
+   */
+  private void addToQueue(String payload) {
+    waitForPayloadSend();
+
+    synchronized (payloads) {
+      payloads.add(payload);
+    }
+
+    Logger.LOG_DEBUG("payload completed count is: " + payloads.size());
+    if (payloads.size() >= MAX_PAYLOADS) {
+      status = STATUS_STOP;
+    }
+  }
+
+  /**
+   * Gets the next payload from the top of the payload queue.
+   *
+   * @return next payload
+   */
+  public String getNextPayload() {
+    String val = "";
+    int topOfQueue = 0;
+
+    synchronized (payloads) {
+      if (!payloads.isEmpty()) {
+        val = (String) payloads.remove(topOfQueue);
+        if (status == STATUS_STOP) {
+          status = STATUS_OKAY;
+        }
+      }
+    }
+
+    return val;
+  }
+
+  /**
+   * Clear the contents of the payload queue.
+   */
+  public void clearQueue() {
+    synchronized (payloads) {
+      payloads.clear();
+    }
+  }
+
+  /**
+   * Only store up to {@link #MAX_PAYLOADS} payloads for OSIsoft to send out. If there is a full
+   * queue, wait for payloads to be sent out before adding more.
+   */
+  private void waitForPayloadSend() {
+
+    // Wait until status is not STOP
+    while (status == STATUS_STOP) {
+      if (payloads.size() != MAX_PAYLOADS) {
+        status = STATUS_OKAY;
+      }
+      try {
+        Thread.sleep(WAIT_FOR_PAYLOAD_SEND_MILLISECONDS);
+      } catch (InterruptedException e) {
+        Logger.LOG_EXCEPTION(e);
```

Comment:
you should add another log message to indicate that  an error has occurred. Its very unlikely someone will have their log settings set to capture exception messages.